text/1120-deprecate-initializers.md

```diff
@@ -0,0 +1,307 @@
+
+1. **No async/await support**: Initializers cannot be awaited, making asynchronous setup operations difficult to coordinate and debug.
+
+2. **Implicit execution order**: The `before` and `after` properties create implicit dependencies that are hard to reason about and can lead to subtle ordering bugs.
+
+3. **Hidden lifecycle**: The automatic execution of initializers makes the application boot process opaque and harder to debug.
+
+4. **Limited access to application state**: Initializers receive only the application or instance object, without access to other important lifecycle information.
+
+The replacement functionality provides explicit control over the application lifecycle by allowing developers to write setup code directly in their application entry point, with full access to modern JavaScript async/await patterns.
+
+## Transition Path
+
+### Current Initializer Usage Patterns
+
+Application initializers are currently defined in `app/initializers/` and run during application boot:
+
+```js
+// app/initializers/session.js
+export default {
+  name: 'session',
+  initialize(application) {
+    application.inject('route', 'session', 'service:session');
+  }
+};
+```
+
+Instance initializers are defined in `app/instance-initializers/` and run during instance creation:
+
+```js
+// app/instance-initializers/current-user.js
+export default {
+  name: 'current-user',
+  initialize(applicationInstance) {
+    let session = applicationInstance.lookup('service:session');
+    return session.loadCurrentUser();
+  }
+};
+```
+
+### New Explicit Lifecycle Pattern
+
+This would replace the contents of the script tag in the existing `@ember/app-blueprint`:
+```html
+<script type="module">
+  import Application from './app/app';
+  import environment from './app/config/environment';
+
+  Application.create(environment.APP);
+</script>
+```
+
+However! We will not change the default from the above, as the simple case is the most common.
+
+> [!IMPORTANT]
+> The above 3 line entrypoint for ember applications will remain the default. This RFC is primarily about updates to documentation showing how we can achieve a new way of "initializers" while also slimming the default experience for new projects.
+
+The new pattern provides explicit control over application lifecycle in the main application entry point:
+
+```js
+import Application from '#app/app.ts';
+import environment from '#app/config.ts';
+
+let app = Application.create({
+  ...environment.APP,
+  autoboot: false,
+});
+
+// anything running here with access to `app`
+// is equivelent to classic "initializers"
+//
+// These can be awaited, which was not possible before
+console.log(app);
+
+await app.boot();
+let instance = await app.buildInstance();
+await instance.boot();
+await instance.visit('/');
+
+// anything running here with access to `instance`
+// is equivelent to classic "instance initializers"
+//
+// These can be awaited, which was not possible before
+console.log(instance);
+```
```

Comment:
Where is that file located? What's the proposed filename?

Second, a thing currently missing (ok, badly located) is to programmatically configure your app (currently `beforeModel()` hook in application route). This seems like the place to do that config as well (when the instance is available then). When thinking about this as runtime configuration, that - to me - brings all that in one place I was hoping for years to have.

Reply:
Ah, sorry, it's the index.html' i'll clarify that

Second, yeah, perhaps i'll add an example for that